workflow/rules/map.smk

```diff
@@ -117,6 +117,8 @@ rule fastq_quality_filter:
         os.path.join(config["local_log"], "fastq_quality_filter_{sample}.log"),
     container:
         "docker://quay.io/biocontainers/fastx_toolkit:0.0.14--H87F3376_10"
+    conda:
+        os.path.join(workflow.basedir, "envs", "fastx_toolkit.yaml")
```

Comment:
Can you please create an issue for using `pathlib.Path` objects instead of manipulating strings for path operations across the Snakemake descriptor files? It's much less verbose, safer and easier to read.

Compare:

```
from pathlib import Path

WF_DIR = Path(workflow.basedir)

...
        WF_DIR / "envs" / "fastx_toolkit.yaml"
```

with

```
import is

        os.path.join(workflow.basedir, "envs", "fastx_toolkit.yaml)
```

The `WF_DIR` instantiation you only need to do once (and another one for `LOG_DIR` and maybe few others), but every one of the many many lines that do path operations will become much easier to manage.